### PR TITLE
[2/N] Fix clang-tidy warnings in python_variable_methods.cpp 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #139007

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: tools/autograd/templates/python_variable_methods.cpp [hunk 1]
```diff
@@ -1186,14 +1186,14 @@ static PyObject* THPVariable_set_(
         " for argument 1 'storage'");
       auto dispatch_set_ = [](const Tensor& self, Storage source) -> Tensor {
         pybind11::gil_scoped_release no_gil;
-        return self.set_(source);
+        return self.set_(std::move(source));
       };
       return wrap(dispatch_set_(self, storage));
```
Comment (Skylion007) [inline tools/autograd/templates/python_variable_methods.cpp:1191]: Actually, should any of these wrap calls also move their args? Might be good for future PR.

Comment (cyyever) [reply tools/autograd/templates/python_variable_methods.cpp:1191]: There will be 3,4,5... until all are fixed.

Comment (cyyever) [reply tools/autograd/templates/python_variable_methods.cpp:1191]: Most importantly, it should be included into linterruner

**Pull request closed without merging**